build_tools/cmake/run_jax_tests.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set -xeo pipefail
+
+pjrt_platform=$1
+
+if [ -z "${pjrt_platform}" ]; then
+    set +x
+    echo "Usage: run_jax_tests.sh <pjrt_platform>"
+    echo "  <pjrt_platform> can be 'cpu', 'cuda', 'rocm' or 'vulkan'"
+    exit 1
+fi
+
+# cd into the PJRT plugin dir
+ROOT_DIR="${ROOT_DIR:-$(git rev-parse --show-toplevel)}"
+cd "${ROOT_DIR}/integrations/pjrt"
+
+# perform some differential testing
+actual_jax_platform=iree_${pjrt_platform}
+expected_jax_platform=cpu
+
+# this function will execute the test python script in
+# both cpu mode and the IREE PJRT mode,
+# and then compare the difference in the output
+diff_jax_test() {
+    local test_py_file=$1
+
+    echo "executing ${test_py_file} in ${expected_jax_platform}.."
+    local expected_tmp_out=$(mktemp /tmp/jax_test_result_expected.XXXXXX)
+    JAX_PLATFORMS=$expected_jax_platform python $test_py_file > $expected_tmp_out
```

Comment:
Hm.. this failed using the default jax cpu platform : https://github.com/iree-org/iree/actions/runs/11948618834/job/33325377235?pr=19222#step:7:64

```
  File "/home/runner/work/iree/iree/.venv/lib/python3.11/site-packages/jax/_src/numpy/lax_numpy.py", line 2070, in array
    out = np.array(object, dtype=dtype, ndmin=ndmin, copy=False)  # type: ignore[arg-type]
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Unable to avoid copy while creating an array as requested.
```

```
Successfully installed iree-compiler-20241104.1068 iree-pjrt-plugin-cpu-0.1.dev1 jaxlib-0.4.35 ml-dtypes-0.5.0 mpmath-1.3.0 numpy-2.1.3 scipy-1.14.1 sympy-1.13.3

Successfully installed jax-0.4.20 jaxlib-0.4.20 opt-einsum-3.4.0
```

Could try with older numpy and/or newer jax?

Reply:
Ahh I think it's due to the version of numpy that jax 0.4.20 depends on. It should be 1.x but it seems 2.0 is installed. I'll fix it.